internal/tester.go

```diff
@@ -79,63 +94,120 @@ func (t *Tester) ExecuteTests() error {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
+	log.Println("Loading default configuration...")
+	configuration, err := kconfig.DefaultConfiguration()
+	if err != nil {
+		return errors.Wrap(err, "failed to load Chainsaw default configuration")
+	}
+	configuration.Spec.Discovery.TestFile = tf
+	configuration.Spec.Execution.Parallel = ptr.To(1)
+	configuration.Spec.Cleanup.SkipDelete = true
+
+	log.Printf("- Using test file: %s\n", configuration.Spec.Discovery.TestFile)
+	log.Printf("- ApplyTimeout %v\n", configuration.Spec.Timeouts.Apply.Duration)
+	log.Printf("- AssertTimeout %v\n", configuration.Spec.Timeouts.Assert.Duration)
+	log.Printf("- CleanupTimeout %v\n", configuration.Spec.Timeouts.Cleanup.Duration)
+	log.Printf("- DeleteTimeout %v\n", configuration.Spec.Timeouts.Delete.Duration)
+	log.Printf("- ErrorTimeout %v\n", configuration.Spec.Timeouts.Error.Duration)
+	log.Printf("- ExecTimeout %v\n", configuration.Spec.Timeouts.Exec.Duration)
+	log.Printf("- Parallel %d\n", *configuration.Spec.Execution.Parallel)
+	color.Init(false, true)
+
+	log.Println("Loading tests...")
+	tests, err := discovery.DiscoverTests(tf, nil, false, t.options.Directory)
+	if err != nil {
+		return errors.Wrap(err, "failed to discover test cases")
+	}
+
+	var testToRun []discovery.Test
+	for _, test := range tests {
+		if test.Err != nil {
+			log.Printf("- %s (%s) - (%s)\n", test.Test.Name, test.BasePath, test.Err)
+		} else {
+			log.Printf("- %s (%s)\n", test.Test.Name, test.BasePath)
+			testToRun = append(testToRun, test)
+		}
+	}
+
+	log.Println("Running tests...")
+	overrides := clientcmd.ConfigOverrides{}
+	restConfig, err := restutils.DefaultConfig(overrides)
+	if err != nil {
+		return errors.Wrap(err, "failed to load Kubernetes config")
+	}
+
+	tc, err := enginecontext.InitContext(configuration.Spec, restConfig, nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize execution context")
+	}
 
-	cmd := exec.CommandContext(ctx, "bash", "-c", chainsawCommand) // #nosec G204
-	stdout, _ := cmd.StdoutPipe()
-	if err := cmd.Start(); err != nil {
-		return errors.Wrapf(err, "cannot start chainsaw: %s", chainsawCommand)
+	clock := clock.RealClock{}
+	onFailure := func() {
+		log.Println("Test failed.")
 	}
 
-	// Start ticker for kubectl command every 30 seconds
+	runner := runner.New(clock, onFailure)
+
 	ticker := time.NewTicker(t.options.LogCollectionInterval)
 	done := make(chan bool)
+	var mutex sync.Mutex
 	defer func() {
 		ticker.Stop()
 		close(done)
 	}()
 
-	var mutex sync.Mutex
 	go logCollector(done, ticker, &mutex, resources)
 
-	sc := bufio.NewScanner(stdout)
-	for sc.Scan() {
-		mutex.Lock()
-		log.Println(sc.Text())
-		mutex.Unlock()
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	if err := runnerflags.SetupFlags(configuration.Spec); err != nil {
+		return err
 	}
```

Comment:
It used to be that the logs of the CLI were printed here in a stream and as they came in, is this the case now? Or are they printed once for a test execution and then all together?

Reply:
does this help to see the log flow in a uptest run?

https://www.loom.com/share/a21158bb0d0945c984ca5f385a26181f